lib/active_model/one_time_password.rb

```diff
@@ -5,20 +5,21 @@ module OneTimePassword
     module ClassMethods
       def has_one_time_password(options = {})
         cattr_accessor :otp_column_name, :otp_counter_column_name
-        class_attribute :otp_digits, :otp_counter_based
+        class_attribute :otp_digits, :otp_counter_based, :skip_secret_generation
 
         self.otp_column_name = (options[:column_name] || "otp_secret_key").to_s
         self.otp_digits = options[:length] || 6
 
         self.otp_counter_based = (options[:counter_based] || false)
         self.otp_counter_column_name = (options[:counter_column_name] || "otp_counter").to_s
+        self.skip_secret_generation = options[:skip_secret_generation] || false
 
         include InstanceMethodsOnActivation
 
         before_create do
```

Comment:
> @robertomiranda Do you mean that otp_enabled will be a column in database? By now I'm just controlling if otp is enabled for users if they have the otp_column filled or not.
I think that's a possible solution but ideally, I prefer not having another column and I'd like to only skip the before_create callback by default for not generating the secret. 

I was thinking of passing the option here:

```suggestion
        before_create(if: condition) do
```

It has not to be another column, it could be a method or even a proc taking advantage that active model already provides this functionality:

https://github.com/rails/rails/blob/aeba121a83965d242ed6d7fd46e9c166079a3230/activerecord/test/cases/callbacks_test.rb#L132-L147

https://github.com/rails/rails/blob/aeba121a83965d242ed6d7fd46e9c166079a3230/activerecord/test/cases/callbacks_test.rb#L402-L420

Reply:
Oh I see, thanks for clarifying things @robertomiranda, wdyt about https://github.com/heapsource/active_model_otp/pull/56/commits/297f0d17137f0fa2fd6b5ff642945729b727e9d2?

Reply:
LGTM 👍